src/services/completions.ts

```diff
@@ -1902,7 +1931,21 @@ namespace ts.Completions {
                 existingMemberNames.set(existingName!, true); // TODO: GH#18217
             }
 
-            return contextualMemberSymbols.filter(m => !existingMemberNames.get(m.escapedName));
+            const filteredSymbols = contextualMemberSymbols.filter(m => !existingMemberNames.get(m.escapedName));
+            setSortTextToMemberDeclaredBySpreadAssignment(membersDeclaredBySpreadAssignment, contextualMemberSymbols);

```

Comment:
Since the result of this is filtered symbols, you want to pass in `filteredSymbols` here instead of `contextualMemberSymbols`.